File: pinyougou-web-manager/src/main/java/cn/itcast/core/controller/order/OrderController.java
package cn.itcast.core.controller.order;

import cn.itcast.core.entity.PageResult;
import cn.itcast.core.pojo.order.Order;
import cn.itcast.core.service.order.OrderService;
import com.alibaba.dubbo.config.annotation.Reference;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Date;

@RestController
@RequestMapping("/order")
public class OrderController {

    @Reference
    private OrderService orderService;

    /**
     *  运营商后台-订单统计
     * @param page
     * @param rows
     * @param createTime
     * @return
     */
    @RequestMapping("/statistics.do")
    public PageResult search(Integer page, Integer rows, String createTime, @RequestBody Order order){
        System.out.println(createTime);
        Date startDate = null;
        Date endDate = null;
        if (createTime!=null&&!"".equals(createTime)){
            startDate = new Date();
            endDate = new Date();
            if ("1".equals(createTime)){
                startDate.setHours(startDate.getHours()-24);
            }else
            if ("2".equals(createTime)){
                startDate.setDate(startDate.getDate()-7);
            }else
            if ("3".equals(createTime)){
                startDate.setMonth(startDate.getMonth()-1);
            }else
            if ("4".equals(createTime)){
                startDate.setMonth(startDate.getMonth()-3);
            }else
            if ("5".equals(createTime)){
                startDate.setMonth(startDate.getMonth()-6);
            }else
            if ("6".equals(createTime)){
                startDate.setMonth(startDate.getMonth()-12);
            }else
            if ("7".equals(createTime)){
                startDate.setMonth(startDate.getMonth()-12);
                endDate.setMonth(endDate.getMonth()-12);
            }
        }
        System.out.println(startDate);
        System.out.println(endDate);
        return orderService.statistics(page, rows, startDate,endDate,order);
    }
    /**
     *  运营商后台-订单查询
     * @param page
     * @param rows
     * @param minPrice
     * @param maxPrice
     * @return
     */
    @RequestMapping("/search.do")
    public PageResult search(Integer page, Integer rows, String minPrice ,String maxPrice , @RequestBody Order order){
        return orderService.search(page, rows, minPrice,maxPrice,order);
    }
}

File: pinyougou-web-manager/src/main/webapp/js/controller/orderController.js
 //控制层 
app.controller('orderController' ,function($scope,$controller,orderService){
	
	$controller('baseController',{$scope:$scope});//继承
	
    //读取列表数据绑定到表单中  
	$scope.findAll=function(){
        orderService.findAll().success(
			function(response){
				$scope.list=response;
			}			
		);
	}    
	
	//分页
	$scope.findPage=function(page,rows){
        orderService.findPage(page,rows).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}
	
	//查询实体 
	$scope.findOne=function(id){
        orderService.findOne(id).success(
			function(response){
				$scope.entity= response;					
			}
		);				
	}
	
	$scope.searchEntity={};//定义搜索对象

	$scope.searchEntity.creatTimeStr=$filter("date")($scope.searchEntity.creatTimeStr,"yyyy-MM-dd")
	
	//搜索
	$scope.search=function(page,rows){
        /*orderService.statistics(page,rows,$scope.createTime,$scope.searchEntity).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);*/

        orderService.search(page,rows,$scope.minPrice,$scope.maxPrice,$scope.searchEntity).success(
            function(response){
                $scope.list=response.rows;
                $scope.paginationConf.totalItems=response.total;//更新总记录数
            }
        );



	}


    
	// 显示状态
	$scope.status = ["未审核","审核通过","审核未通过","关闭"];
});
